docs/v1beta1-checklist-developer-guide.md

```diff
@@ -0,0 +1,126 @@
+and cannot be modified. However, it will inform how all other components are
+implemented, so it is important to have a strong understanding of both the
+fields it exposes and how they are used. Many cloud provider's will use default
+values for omitted fields that may not be intuitive or equal to the [zero-value]
+of the Go type they are implemented in by the SDK. It is usually not enough to
+just read the code exposed by the API. Authors of `beta` Crossplane APIs should
+be familiar with how the cloud provider responds to and modifies each field
+either by reading additional documentation or testing different configurations
+directly against the API.
+
+- [ ] I understand the behavior of any value provided to every field for the
+  cloud provider API type.
+- [ ] I understand which values are configurable and which simply report the
+  current status of the resource.
+
+## Crossplane API Type
+
+The Crossplane API type is a Kubernetes representation of the cloud provider
+API. All `beta` Crossplane API types should be a [high fidelity] representation
+of the cloud provider API type. This means that any functionality that can be
+achieved using the cloud provider API should also be possible by creating the
+Crossplane representation of that API type. For full detail about how to design
+a Crossplane API type, please see our [API patterns design doc].
+
+- [ ] I have read and understand the Crossplane [API patterns design doc].
+- [ ] I have removed any field that is related to naming the external resource,
+  as it will be handled via the `crossplane-runtime` [external name annotation].
+- [ ] I have added all fields that can be configured at any point in the
+  lifetime of the external resource to the `spec.forProvider` struct of the
+  Crossplane API type.
+- [ ] I have added all fields that cannot be configured at any point in the
+  lifetime of the external resource to the `status.atProvider` struct of the
+  Crossplane API type.
+- [ ] I have provided a code comment on every field that is descriptive of its
+  functionality.
+- [ ] I have denoted all *optional* fields in `spec.forProvider` with a `//
+  +optional` tag and made them a pointer type.
+- [ ] I have denoted all fields in `spec.forProvider` that cannot be modified
+  after initial external resource creation with a `// +immutable` tag.
+- [ ] I have added an additional `_____Ref` field for any field that is used to
+  reference another external resource for which there is currently a Crossplane
+  API type representation. I have also added the necessary [attribute
+  referencer] methods in accordance with the Crossplane [cross-resource
+  reference design document]. 
+- [ ] I have replaced any fields that require [sensitive input] (e.g. password
+  for a database) with a `_____SecretRef` field in accordance with the API
+  patterns design doc.
+
+In addition to this checklist, it is recommended that an author also looks at
```

Comment:
Linking one example that we are committed to keeping high quality could be good here, maybe `CloudSQLInstance`?